src/core/router/transports/ssu/session.cc

```diff
@@ -620,56 +621,71 @@ void SSUSession::ProcessRelayRequest(
   SendRelayIntro(session.get(), from);
 }
 
-// TODO(anonimal): bytestream refactor
 void SSUSession::SendRelayRequest(
-    std::uint32_t introducer_tag,
-    const std::uint8_t* introducer_key) {
-  auto address = context.GetRouterInfo().GetSSUAddress();
-  if (!address) {
-    LOG(error)
-      << "SSUSession:" << GetFormattedSessionInfo()
-      << __func__ << ": SSU is not supported";
-    return;
-  }
-  std::array<std::uint8_t, 96 + SSUSize::BufferMargin> buf{
-      {}};  // TODO(unassigned): document size values
-  auto payload = buf.data() + SSUSize::HeaderMin;
-  core::OutputByteStream::Write<std::uint32_t>(payload, introducer_tag);
-  payload += 4;
-  *payload = 0;  // no address
-  payload++;
-  core::OutputByteStream::Write<std::uint16_t>(payload, 0, false);  // port = 0
-  payload += 2;
-  *payload = 0;  // challenge
-  payload++;
-  memcpy(payload, (const std::uint8_t *)address->key, 32);
-  payload += 32;
-  core::OutputByteStream::Write<std::uint32_t>(
-      payload, core::Rand<std::uint32_t>());  // nonce
-  std::array<std::uint8_t, SSUSize::IV> iv;
-  kovri::core::RandBytes(iv.data(), iv.size());
-  auto relay_request = SSUPayloadType::RelayRequest;
-  if (m_State == SessionState::Established) {
-    FillHeaderAndEncrypt(
-        relay_request,
-        buf.data(),
-        96,
-        m_SessionKey,
-        iv.data(),
-        m_MACKey);
-  } else {
-    FillHeaderAndEncrypt(
-        relay_request,
-        buf.data(),
-        96,
-        introducer_key,
-        iv.data(),
-        introducer_key);
-  }
+    const std::uint32_t introducer_tag,
+    const std::uint8_t* introducer_key)
```

Comment:
Noob question about const-correctness: is it a good idea to make `introducer_key` `const uint8_t* const`?

When is it necessary to also make the pointer const?

Reply:
>is it a good idea to make introducer_key const uint8_t* const

That's not even correct syntax - and won't even build. Even if it could, it's not necessary. 

>When is it necessary to also make the pointer const?

lmgtfy.